### PR TITLE
add lang#sml layer

### PR DESCRIPTION
**Pull request by thawk**
### PR Prelude

Thank you for working on SpaceVim! :)

Please complete these steps and check these boxes before filing your PR:

- [x] I have read and understood SpaceVim's [CONTRIBUTING](https://github.com/SpaceVim/SpaceVim/blob/master/CONTRIBUTING.md) document.
- [x] I have read and understood SpaceVim's [CODE_OF_CONDUCT](https://github.com/SpaceVim/SpaceVim/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I understand my PR may be closed if it becomes obvious I didn't actually perform all of these steps.

### Why this change is necessary and useful?

I found that there's a PR for lang#sml, but it's not complete.  So I think maybe someone is interest in Stand ML.

### CODE REVIEW

File: autoload/SpaceVim/layers/lang/sml.vim [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+function! SpaceVim#layers#lang#sml#set_variable(var) abort
+  let g:sml_smlnj_executable = get(a:var, 'smlnj_path', 'sml')
+  let g:sml_mlton_executable = get(a:var, 'mlton_path', 'mlton')
+  let s:sml_repl_options = get(a:var, 'repl_options', s:sml_repl_options)
+  let g:sml_repl_options = s:sml_repl_options
+  let g:sml_auto_create_def_use = get(a:var, 'auto_create_def_use', 'mlb')
+  let g:sml_greek_tyvar_show_tick = get(a:var, 'enable_conceal_show_tick', '0')
+  let s:sml_enable_conceal = get(a:var, 'enable_conceal', s:sml_enable_conceal)
+  let s:sml_file_head = get(a:var, 'file_head', s:sml_file_head)
```
Comment (wsdjeg) [inline autoload/SpaceVim/layers/lang/sml.vim]: in `lang#lua` and `lang#python`, we use `lua_file_head` and `python_file_head` as the layer option name, I think maybe we should we `sml_file_head`.

Comment (thawk) [reply autoload/SpaceVim/layers/lang/sml.vim]: updated to ``sml_file_head``.
I found out that ``python`` layer use both ``python-file-head`` and ``python_file_head``, but ``ruby`` and ``sh`` only support the ``ruby-file-head`` and ``bash-file-head``, maybe we should correct them.

Comment (wsdjeg) [reply autoload/SpaceVim/layers/lang/sml.vim]: we should use sml_file_head, python-file-head is just for backward compatibility.

**Comment by wsdjeg** [comment]
merged, thanks.

**Pull request merged**